notifier/notifier.go

```diff
@@ -0,0 +1,103 @@
+package notifier
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+)
+
+const (
+	// IOS identifier for an ios mnotification.
+	IOS = 1
+	// Android identifier for an android notification.
+	Android = 2
+)
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Notifier handles android and ios push notifications.
+type Notifier struct {
+	client httpClient
```

Comment:
Any reason for `httpClient` type instead of `*http.Client`?

Reply:
[httpClient](https://github.com/status-im/status-go/pull/910/files#diff-4063493f3b3cf719235a092fd180dd56R20) is an interface :)

Reply:
Right, but I just wonder what was your motivation to create this interface and use it instead of providing a concrete type.

Reply:
ah, sorry, again that's missing from a previous approach 😓 
fixed now